app/controllers/surveys_controller.rb

```diff
@@ -1,4 +1,58 @@
 class SurveysController < ApplicationController
   def send_survey
   end
+
+  def create
+    @survey = Survey.create!(survey_params)
+    questions = []
+
+    params[:questions].each do |question_params|
+      question = Question.create!(
+        text: question_params[:text],
+        question_type: question_params[:question_type].downcase,
+        number: question_params[:id],
+      )
+      questions.push(question)
+
+      question_params[:options].each do |response_params|
+        ResponseChoice.create!(
+          key: response_params[:key].downcase,
+          text: response_params[:text],
+        )
+      end
+    end
+
+    @survey.update!(first_question: questions.find{ |q| q.number = 1 })
+
+    params[:questions].each do |question_params|
+      if (question_params[:default_next_question_id].to_i == -1)
+        question_params[:options].each do |response_params|
+          ConditionalQuestionOrder.create!(
+            survey: @survey,
+            question: questions[question_params[:id].to_i - 1],
+            response_choice: questions[question_params[:id].to_i - 1].response_choices.find{ |r| r.key = response_params[:key] },
+            next_question: response_params[:next_question_id].to_i > 0 ? questions[response_params[:next_question_id].to_i - 1] : nil
```

Comment:
Can we encapsulate `question_params[:id].to_i -1` and `response_params[:next_question_id].to_i` into a variable since you call it twice and it makes it hard to read